tmk_core/protocol/lufa/lufa.c

```diff
@@ -288,6 +292,84 @@ static void Console_Task(void)
 }
 #endif
 
+/*******************************************************************************
+ * Joystick
+ ******************************************************************************/
+#ifdef JOYSTICK_ENABLE
+
+typedef struct {
+    #if JOYSTICK_AXES_COUNT>0
+    int8_t  axes[JOYSTICK_AXES_COUNT];
+    #endif
+    
+    #if JOYSTICK_BUTTON_COUNT>0
+    uint8_t buttons[(JOYSTICK_BUTTON_COUNT-1)/8+1];
+    #endif
+} __attribute__ ((packed)) joystick_report_t;
+
+void send_joystick_packet(joystick_t* joystick){
+    
+    uint8_t timeout = 255;
+    uint8_t where = where_to_send();
+    
+    if (where != OUTPUT_USB && where != OUTPUT_USB_AND_BT) {
+      return;
+    }
+    
+    joystick_report_t r = {
```

Comment:
This code is very similar to the one in vusb.c, anyway to avoid some of the code repetition?